mobius/src/geometry/ray.rs

```diff
@@ -13,6 +15,19 @@ impl Ray {
         ComplexError::require_finite("start", start)?;
         Ok(Self { start, unit_dir })
     }
+
+    pub fn to_primitive(&self) -> RenderPrimitive {
```

Comment:
now that I changed the definition of `Renderable`, this is just a `render()` function that succeeds